packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,273 @@
+    sellAsset,
+    buyAsset,
+    affiliateBps,
+    receiveAddress,
+    accountNumber,
+    sendAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    slippageTolerancePercentageDecimal,
+  } = input
+
+  const { assetsById } = deps
+
+  const jupiterUrl = deps.config.REACT_APP_JUPITER_API_URL
+
+  if (accountNumber === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `accountNumber is required`,
+        code: TradeQuoteError.UnknownError,
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: `sendAddress is required`,
+        code: TradeQuoteError.UnknownError,
+      }),
+    )
+  }
+
+  const maybePriceResponse = await getJupiterPrice({
+    apiUrl: jupiterUrl,
+    sourceAsset: sellAsset.assetId === solAssetId ? wrappedSolAssetId : sellAsset.assetId,
+    destinationAsset: buyAsset.assetId === solAssetId ? wrappedSolAssetId : buyAsset.assetId,
+    commissionBps: affiliateBps,
+    amount: sellAmount,
+    slippageBps: convertDecimalPercentageToBasisPoints(
+      slippageTolerancePercentageDecimal ??
+        getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Jupiter),
+    ).toFixed(),
+  })
+
+  if (maybePriceResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybePriceResponse.unwrap()
+
+  const contractAddress =
+    buyAsset.assetId === solAssetId ? undefined : fromAssetId(buyAsset.assetId).assetReference
+
+  const adapter = deps.assertGetSolanaChainAdapter(sellAsset.chainId)
+
+  const isCrossAccountTrade = receiveAddress ? receiveAddress !== sendAddress : false
+
+  const { instruction: createTokenAccountInstruction, destinationTokenAccount } =
+    contractAddress && isCrossAccountTrade
+      ? await adapter.createAssociatedTokenAccountInstruction({
+          from: sendAddress,
+          to: receiveAddress!,
+          tokenId: contractAddress,
+        })
+      : { instruction: undefined, destinationTokenAccount: undefined }
+
+  const maybeSwapResponse = await getJupiterSwapInstructions({
+    apiUrl: jupiterUrl,
+    fromAddress: sendAddress,
+    toAddress: isCrossAccountTrade ? destinationTokenAccount?.toString() : undefined,
+    rawQuote: quoteResponse,
+    wrapAndUnwrapSol: buyAsset.assetId === wrappedSolAssetId ? false : true,
```

Comment:
This can also go away now that wrapped SOL is unsupported 
```suggestion
    wrapAndUnwrapSol: true,
```

Reply:
Done in [561cd08](https://github.com/shapeshift/web/pull/8120/commits/561cd0854969e274843124d716a24964dec5892e)